tasks/pdp/task_prove.go

```diff
@@ -631,6 +636,76 @@ func (p *ProveTask) getSenderAddress(ctx context.Context) (common.Address, error
 	return address, nil
 }
 
+func (p *ProveTask) cleanupDeletedRoots(ctx context.Context, proofSetID int64, pdpVerifier *contract.PDPVerifier) error {
+
+	removals, err := pdpVerifier.GetScheduledRemovals(nil, big.NewInt(proofSetID))
+	if err != nil {
+		return xerrors.Errorf("failed to get scheduled removals: %w", err)
+	}
+
+	// Execute cleanup in a transaction
+	ok, err := p.db.BeginTransaction(ctx, func(tx *harmonydb.Tx) (bool, error) {
+
+		for _, removeID := range removals {
+			log.Infow("cleanupDeletedRoots", "removeID", removeID)
+			// Get the pdp_pieceref ID for the root before deleting
+			var pdpPieceRefID int64
+			err := tx.QueryRow(`
+                SELECT pdp_pieceref 
+                FROM pdp_proofset_roots 
+                WHERE proofset = $1 AND root_id = $2
+            `, proofSetID, removeID.Int64()).Scan(&pdpPieceRefID)
+			if err != nil {
+				if errors.Is(err, sql.ErrNoRows) {
+					// Root already deleted, skip
+					continue
+				}
+				return false, xerrors.Errorf("failed to get piece ref for root %d: %w", removeID, err)
+			}
+
+			// Get parked piece ref
+			var parkedPieceRef int64
+			err = tx.QueryRow(`
+				SELECT piece_ref
+				FROM pdp_piecerefs
+				WHERE id = $1
+			`, pdpPieceRefID).Scan(&parkedPieceRef)
+			if err != nil {
+				return false, xerrors.Errorf("failed to get parked piece ref for root %d: %w", removeID, err)
+			}
+
+			// Delete the parked piece ref, this will cascade to the pdp piece ref too
+			_, err = tx.Exec(`
+				DELETE FROM parked_piece_refs
+				WHERE ref_id = $1
+			`, parkedPieceRef)
+			if err != nil {
+				return false, xerrors.Errorf("failed to delete parked piece ref %d: %w", parkedPieceRef, err)
+			}
```

Comment:
pdp.pdp_pieceref is already a pieceref

parked_piece_refs maps "Piece Refs" to deduplicated "Piece IDs" where there is generally one "PieceID" for each piece-cid (though there can occasionally be duplicates, e.g. when pieces aren't verified to have correct hashes yet)

```suggestion
			// Delete the parked piece ref, this will cascade to the pdp piece ref too
			_, err = tx.Exec(`
				DELETE FROM parked_piece_refs
				WHERE ref_id = $1
			`, pdpPieceRefID)
			if err != nil {
				return false, xerrors.Errorf("failed to delete parked piece ref %d: %w", parkedPieceRef, err)
			}
```